chatbot-core/database.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from dataclasses import dataclass, field
+from typing import List
+from uuid import uuid4
+
+from qdrant_client import QdrantClient
+from qdrant_client.http.exceptions import UnexpectedResponse
+from qdrant_client.http.models import Distance, VectorParams
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class QdrantDatabase:
+    """Qdrant Vector Database"""
+
+    collection_name: str
+    host: str = "localhost"
+    port: int = 6333
+    db_client: QdrantClient = field(init=False)
+
+    def __post_init__(self):
+        """Initialize database client"""
+        self.db_client = QdrantClient(host=self.host, port=self.port)
+
+    def _collection_exists(self, collection_name: str) -> bool:
+        """Check if the collection in db already exists"""
+        try:
+            self.db_client.get_collection(collection_name=collection_name)
+            return True
+        except UnexpectedResponse:
+            return False
+
+    def set_collection(self, vector_size: int = 384):
```

Comment:
Does changing vector_size require collection regeneration?

Reply:
Yes, it does.

Reply:
Let's define this on settings.py as hard-coded, and let's add a note on how changing this can have an impact on existing data.